src/test/kotlin/org/onflow/flow/sdk/crypto/CryptoTest.kt

```diff
@@ -107,10 +107,170 @@ internal class CryptoTest {
+    @Test
+    fun `Sanity check KMAC128`() {
+        val input = byteArrayOf(0x00, 0x01, 0x02, 0x03)
+        val expected = listOf(
+            hexStringToByteArray("E5780B0D3EA6F7D3A429C5706AA43A00FADBD7D49628839E3187243F456EE14E"),
+            hexStringToByteArray("3B1FBA963CD8B0B59E8C1A6D71888B7143651AF8BA0A7070C0979E2811324AA5")
+        )
+        val key = hexStringToByteArray("404142434445464748494A4B4C4D4E4F505152535455565758595A5B5C5D5E5F")
+        val customizers = listOf("".toByteArray(), "My Tagged Application".toByteArray())
+        val outputSize = 32
+
+        customizers.forEachIndexed { index, customizer ->
+            // Test full input processing
+            val hasher1 = HasherImpl(HashAlgorithm.KMAC128, key, customizer, outputSize)
+            val hash1 = hasher1.hash(input)
+            assertArrayEquals(expected[index], hash1)
+
+            // Test incremental input processing
+            val hasher2 = HasherImpl(HashAlgorithm.KMAC128, key, customizer, outputSize)
+            hasher2.update(input, 0, 2)
+            hasher2.update(input, 2, input.size - 2)
+            val hash2 = hasher2.doFinal(outputSize)
+            assertArrayEquals(expected[index], hash2)
+        }
+    }
+
+    @Test
+    fun `Test short key length`() {
+        val input = byteArrayOf(0x00, 0x01, 0x02, 0x03)
+        val key = hexStringToByteArray("404142434445464748494A4B4C4D4E4F505152535455565758595A5B5C5D5E5F")
+        val customizer = "".toByteArray()
+        val outputSize = 32
+
+        val exception = assertThrows(IllegalArgumentException::class.java) {
+            HasherImpl(HashAlgorithm.KMAC128, key.sliceArray(0..<15), customizer, outputSize).hash(input)
+        }
+        assertEquals("KMAC128 requires a key of at least 16 bytes", exception.message)
+    }
+
+    @Test
+    fun `Test nil output size`() {
+        val input = byteArrayOf(0x00, 0x01, 0x02, 0x03)
+        val key = hexStringToByteArray("404142434445464748494A4B4C4D4E4F505152535455565758595A5B5C5D5E5F")
+        val customizer = "".toByteArray()
+
+        val exception = assertThrows(IllegalArgumentException::class.java) {
+            HasherImpl(HashAlgorithm.KMAC128, key, customizer, 0).hash(input)
+        }
+        assertEquals("Output size must be at least 256 bits (32 bytes)", exception.message)
+    }
+
+    @Test
+    fun `Test non-nil shorter output size than 32 bytes`() {
+        val input = byteArrayOf(0x00, 0x01, 0x02, 0x03)
+        val key = hexStringToByteArray("404142434445464748494A4B4C4D4E4F505152535455565758595A5B5C5D5E5F")
+        val customizer = "".toByteArray()
+        val outputSize = 16
+
+        val exception = assertThrows(IllegalArgumentException::class.java) {
+            HasherImpl(HashAlgorithm.KMAC128, key, customizer, outputSize).hash(input)
+        }
+        assertEquals("Output size must be at least 256 bits (32 bytes)", exception.message)
+    }
+
+   @Test
+    fun `Test output size longer than 32 bytes`() {
```

Comment:
Looking into this one, using sample 4 from https://csrc.nist.gov/CSRC/media/Projects/Cryptographic-Standards-and-Guidelines/documents/examples/Kmac_samples.pdf with output size 64 but am getting an error currently

Reply:
Sample 4 from the test vector is using 256-bits security, which means it's based on shake256 underneath. However the implementation in the repo is based on shake128, you can see it is specified [here](https://github.com/onflow/flow-jvm-sdk/blob/hashing-algorithms/src/main/kotlin/org/onflow/flow/sdk/crypto/Crypto.kt#L178). So it's expected to find different results than the test vector. 

I don't think we should be using sample 4 because we don't support KMAC256. Since there is no official test vector of KMA128 with an output larger than 32 bytes, we could test against another trusted library (by hardcoding the expected result), I can look around for such data.

Reply:
I didn't find "official" test vectors but I crossed checked the KMAC128 implementation of this [python lib](https://pycryptodome.readthedocs.io/en/v3.20.0/src/introduction.html) against the Flow Go implementation in [this repo](https://github.com/onflow/crypto/) (which is based on CSHAKE from the standard Go lib) and I confirm both give the same results:

Based on the same inputs from sample #1 and sample #2 from https://csrc.nist.gov/CSRC/media/Projects/Cryptographic-Standards-and-Guidelines/documents/examples/Kmac_samples.pdf (that you already integrated), using a 48-bytes output (instead of 32) should output:
- sample #1: `0x4f5967393bd357c13cf1b0aff13c2abe075dd68edee33d6b8cb06f5b2a4d5232c11b439f3c20b20a4f04b0549d9caa10`
- sample #2: `0x99d9364ab5d2b748cb843b27f5a4f4fb06ea87306fa141a676cbb4b39c5c5f12d36b7b76aeea81c4f5876e16e6783e72`

Reply:
I confirm your code works great for the new test vector above. I implemented it in this PR https://github.com/onflow/flow-jvm-sdk/pull/69. Feel free to review and merge (or cherry pick)